src/CSnakes.Runtime/Locators/ManagedPythonLocator.cs

```diff
@@ -0,0 +1,215 @@
+﻿using System.Formats.Tar;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.Logging;
+using ZstdSharp;
+
+namespace CSnakes.Runtime.Locators;
+
+internal class ManagedPythonLocator(ILogger logger) : PythonLocator
+{
+    private const string standaloneRelease = "20250106";
+    private static readonly Version defaultVersion = new(3, 12, 8, 0);
+    protected override Version Version { get; } = defaultVersion;
+
+    protected override string GetPythonExecutablePath(string folder, bool freeThreaded = false)
+    {
+        string suffix = freeThreaded ? "t" : "";
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            return Path.Combine(folder, $"python{suffix}.exe");
+        }
+
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+        {
+            return Path.Combine(folder, "bin", $"python{Version.Major}.{Version.Minor}{suffix}");
+        }
+
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+        {
+            return Path.Combine(folder, "bin", $"python{Version.Major}.{Version.Minor}{suffix}");
+        }
+
+        throw new PlatformNotSupportedException($"Unsupported platform: '{RuntimeInformation.OSDescription}'.");
+    }
+
+    public override PythonLocationMetadata LocatePython()
+    {
+        var downloadPath = Path.Join(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "CSnakes", $"python{Version.Major}.{Version.Minor}");
```

Comment:
AppData or Temp?

Reply:
AppData. It uses temp for the tar.zst file then deletes it. Multiple apps using csnakes should use the same downloaded copy